packages/aws-cdk-lib/aws-secretsmanager/lib/rotation-schedule.ts

```diff
@@ -126,18 +128,32 @@ export class RotationSchedule extends Resource {
       );
     }
 
-    let automaticallyAfterDays: number | undefined = undefined;
-    if (props.automaticallyAfter && props.automaticallyAfter.toDays() > 1000) {
-      throw new Error(`automaticallyAfter must not be greater than 1000 days, got ${props.automaticallyAfter.toDays()} days`);
-    }
-    if (props.automaticallyAfter?.toMilliseconds() !== 0) {
-      automaticallyAfterDays = props.automaticallyAfter?.toDays() || 30;
+    let automaticallyAfterDays: number | undefined;
+    let scheduleExpression: string | undefined;
+    if (props.automaticallyAfter) {
+      const automaticallyAfterMillis = props.automaticallyAfter.toMilliseconds();
+      if (automaticallyAfterMillis > 0) {
+        if (automaticallyAfterMillis < Duration.hours(4).toMilliseconds()) {
+          throw new Error(`automaticallyAfter must not be smaller than 4 hours, got ${props.automaticallyAfter.toHours()} hours`);
+        }
+        if (automaticallyAfterMillis > Duration.days(1000).toMilliseconds()) {
+          throw new Error(`automaticallyAfter must not be greater than 1000 days, got ${props.automaticallyAfter.toDays()} days`);
+        }
+        if (props.automaticallyAfter.toHours() >= 24) {
+          automaticallyAfterDays = props.automaticallyAfter.toDays();
+        } else {
+          scheduleExpression = Schedule.rate(props.automaticallyAfter).expressionString;
+        }
+      }
+    } else {
+      automaticallyAfterDays = 30;
     }
 
-    let rotationRules: CfnRotationSchedule.RotationRulesProperty | undefined = undefined;
-    if (automaticallyAfterDays !== undefined) {
+    let rotationRules: CfnRotationSchedule.RotationRulesProperty | undefined;
+    if (automaticallyAfterDays !== undefined || scheduleExpression !== undefined) {
       rotationRules = {
         automaticallyAfterDays,
+        scheduleExpression,
```

Comment:
@lpizzinidev I don't like this implementation because we are returning both `automaticallyAfterDays` and `scheduleExpression` knowing full well that one is undefined. We're then relying on this information elsewhere, which is okay for now, but is an assumption we'll have to maintain forever.

Instead, I think we should just localize on `scheduleExpression`. We can keep the `automaticallyAfter` prop, but wire it to `scheduleExpression` all the time. We can specify `rate(5 days)` if we get a unit in days, or `rate(4 hours)` if the unit is in hours.

`scheduleExpression` can do everything `automaticallyAfterDays` can do, and more, so lets standardize on that.